tests/helpers/test_schema_validation.py

```diff
@@ -0,0 +1,49 @@
+"""The tests for helper config validation."""
+
+from homeassistant.helpers.config_validation import TRIGGER_SCHEMA
+
+
+async def test_nested_trigger_list() -> None:
```

Comment:
Please add a case which show we don't flatten if `triggers` is not the only key

Reply:
Wouldn't that be a validation failure, as there's no current valid trigger we can write that would have that form?